src/web.rs

```diff
@@ -35,9 +39,32 @@ impl WebDisplayImpl {
     }
 }
 
-pub struct WebImpl {
-    canvas: HtmlCanvasElement,
-    ctx: CanvasRenderingContext2d,
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_name = OffscreenCanvasRenderingContext2D)]
+    pub type OffscreenCanvasRenderingContext2d;
+
+    #[wasm_bindgen(catch, method, structural, js_class = "OffscreenCanvasRenderingContext2D", js_name = putImageData)]
+    fn put_image_data(
+        this: &OffscreenCanvasRenderingContext2d,
+        imagedata: &ImageData,
+        dx: f64,
+        dy: f64,
+    ) -> Result<(), JsValue>;
+
+    #[wasm_bindgen(method, structural, js_class = "OffscreenCanvasRenderingContext2D")]
+    pub fn commit(this: &OffscreenCanvasRenderingContext2d);
+}
```

Comment:
`web-sys` doesn't currently support `OffscreenCanvasRenderingContext2D`, a PR has already been made and merged, but it will require a new version release: https://github.com/rustwasm/wasm-bindgen/pull/3319.

Reply:
Let's wait for that, then

Reply:
Alright, will put it into draft mode then.